shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1333,108 +1334,170 @@ - (void)keyboardWillShowNotification:(NSNotification*)notification {
   [self startKeyBoardAnimation:duration];
 }
 
-- (void)keyboardWillChangeFrame:(NSNotification*)notification {
-  // Immediately prior to a change in keyboard frame, this notification is triggered.
-  // There are some cases where UIKeyboardWillShowNotification & UIKeyboardWillHideNotification
-  // do not act as expected and this is used to catch those cases.
+- (BOOL)shouldIgnoreKeyboardNotification:(NSNotification*)notification
+                        notificationName:(NSNotificationName)notificationName {
+  BOOL isKeyboardNotificationForThisView = [self isKeyboardNotificationForThisView:notification];
+  BOOL isKeyboardRotated = [self isKeyboardRotated:notification];
 
-  NSDictionary* info = [notification userInfo];
+  // Don't ignore UIKeyboardWillHideNotification notifications.
+  if (notificationName == UIKeyboardWillHideNotification) {
+    // Skip hide notification when rotation in progress unless triggered by another app.
+    if (isKeyboardRotated && isKeyboardNotificationForThisView) {
+      return YES;
+    }
+    return NO;
+  }
+
+  // Ignore notification when keyboard's dimensions and position are all zeroes,
+  // for UIKeyboardWillChangeFrameNotification. This happens when keyboard is dragged.
+  NSDictionary* info = notification.userInfo;
+  CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
+  if (notificationName == UIKeyboardWillChangeFrameNotification &&
+      CGRectEqualToRect(keyboardFrame, CGRectZero)) {
+    return YES;
+  }
+
+  // Don't ignore other times a keyboard's height or width is 0.
```

Comment:
When does this happen?

Reply:
So the main case this is trying to catch is:
1. When you have the app in Split View, open keyboard on the app, click input field on other app, switch to slide over view, it will send an event (that's triggered by the other app) with a keyboard's height of 0.
A couple of other cases it'll match are:
2. When you switch to predictive-only keyboard from docked/undocked/split
3. When predictive-only/minimized keyboard is dragged

But basically the idea is that if the other app has a notification where the keyboard's height is 0, don't skip it so that the inset gets set to 0.

I'll update the comment to make it make it more clear

Reply:
So I've actually decided to remove the rotation stuff for now. It was an effort to make it look a little better and more like native. However, after more testing I found different natives (like SwiftUI vs keyboardLayoutGuide) handled it differently so I think it needs more time and consideration so I took it out for now and will file a separate issue.